src/main/java/com/epam/xmlparsing/parser/DomParser.java

```diff
@@ -0,0 +1,117 @@
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DomParser implements Parser {
+
+    private static final Logger LOGGER = LogManager.getLogger(DomParser.class);
+
+    private static final String PLANTS_TAG = "plants";
+    private static final String FLOWER_TAG = "flower";
+    private static final String TREE_TAG = "tree";
+    private static final String ID_TAG = "id";
+    private static final String ORIGIN_TAG = "origin";
+    private static final String NAME_TAG = "name";
+    private static final String SOIL_TAG = "soilType";
+    private static final String TIPS_TAG = "growingTips";
+    private static final String NEEDED_TEMPERATURE_TAG = "neededTemperature";
+    private static final String NEEDS_LIGHT_TAG = "needsLight";
+    private static final String WEEKLY_REQUIRED_WATER_TAG = "weeklyRequiredWater";
+    private static final String PETAL_NUMBER_TAG = "petalNumber";
+    private static final String TRUNK_DIAMETER_TAG = "trunkDiameter";
+
+    private void buildPlantFields(Element plantElement, Plant.PlantBuilder builder) {
+        String idLine = plantElement.getAttribute(ID_TAG);
+        int id = Integer.parseInt(idLine);
+        builder.setId(id);
+        String origin = plantElement.getAttribute(ORIGIN_TAG);
+        builder.setOrigin(origin);
+        String name = getElementTextContent(plantElement, NAME_TAG);
+        builder.setName(name);
+        String soilTypeLine = getElementTextContent(plantElement, SOIL_TAG);
+        Soil soilType = Soil.valueOf(soilTypeLine);
+        builder.setSoilType(soilType);
+        NodeList tipsNodes = plantElement.getElementsByTagName(TIPS_TAG);
+        Element tipsNode = (Element) tipsNodes.item(0);
+        String neededTemperatureLine = getElementTextContent(tipsNode, NEEDED_TEMPERATURE_TAG);
+        int neededTemperature = Integer.parseInt(neededTemperatureLine);
+        builder.setNeededTemperature(neededTemperature);
+        String needsLightLine = getElementTextContent(tipsNode, NEEDS_LIGHT_TAG);
+        boolean needsLight = Boolean.parseBoolean(needsLightLine);
+        builder.setNeedsLight(needsLight);
+        String weeklyRequiredWaterLine = getElementTextContent(tipsNode, WEEKLY_REQUIRED_WATER_TAG);
+        int weeklyRequiredWater = Integer.parseInt(weeklyRequiredWaterLine);
+        builder.setWeeklyRequiredWater(weeklyRequiredWater);
+    }
+
+    private Flower buildFlower(Element flowerElement) {
+        LOGGER.info("Started building a flower from found " + FLOWER_TAG + " element");
+        Flower.FlowerBuilder flowerBuilder = new Flower.FlowerBuilder();
+        buildPlantFields(flowerElement, flowerBuilder);
+        String petalNumberLine = getElementTextContent(flowerElement, PETAL_NUMBER_TAG);
+        int petalNumber = Integer.parseInt(petalNumberLine);
```

Comment:
line 68 nd 69 could be in method buildPlantFields, as they are building the flower(fields in your case)